src/index.js

```diff
@@ -1,11 +1,46 @@
-function evaluate(source, endowments = {}) {
-  const names = Object.getOwnPropertyNames(endowments);
-  const s = `(function(endowments) {
-    const { ${names.join(',')} } = endowments;
-    return ${source};
-  })`;
-  // eslint-disable-next-line no-eval
-  return eval(s)(endowments);
-}
-
-module.exports = evaluate;
+// The evaluate maker, which curries the makerOptions.
+export const makeEvaluators = (makerOptions = {}) => {
+  const makeEvaluator = isExpr => (source, endowments = {}, options = {}) => {
+    const fullTransforms = [
+      ...(options.transforms || []),
+      ...(makerOptions.transforms || []),
+    ];
+    const fullEndowments = {
+      ...(makerOptions.endowments || {}),
+      ...endowments,
```

Comment:
just doublechecking.. the local `endowments` will override the ones from `makerOptions`, right?

Reply:
Yes, right.